.github/workflows/main.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [2.7.18]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

Comment:
Ako budeme instalovat requirements? Ako zistim, ktory `server_mode` pouzivam?

Reply:
Nemam este velmi prehlad o testoch. Lisia sa testy pri roznych nastaveniach `server_mode`?

Reply:
Necheme instalovat vsetko rucne ako je vo vzore, ale radsej chceme spustit `setup.py` (ak sa podari), aby logiku setupu sme mali na jednom mieste. Input pre `setup.py` skus poslat nejak podobne ako je v `misc/Dockerfile`

Co sa tyka server modu, zrejme "Local development mode with no email infrastructure". Nepamatam, ci mame nejake testy, ktore zavisia sa server mode. Ked ich budes postupne opravovat, tak pozeraj, ci niektorym netreba iny server mod. Ale tipujem, ze nie, pretoze urcite nechceme, aby testy posielali maily alebo inac interagovali s vonkajsim svetom. Vsetky interakcie s vonkajsim svetom by mali byt namockovane. Aj ked mozno niektore testy nejak s vonkajsim svetom budu potrebovat interagovat. Potom to vyriesime. Zatial asi lokalny mod. Pripadne casom vytvorime novy mod specificky pre testy.